### PR TITLE
Feature/alt tree persistence btc 500

### PR DESCRIPTION
**Pull request by blackyblack**

### CODE REVIEW

File: include/veriblock/blockchain/base_block_tree.hpp [hunk 1]
```diff
@@ -193,7 +193,7 @@ struct BaseBlockTree {
 
   bool operator!=(const BaseBlockTree& o) const { return !operator==(o); }
 
- protected:
+ public:
```
Comment (Warchant) [inline include/veriblock/blockchain/base_block_tree.hpp:196]: why did you change visibility?

File: include/veriblock/blockchain/blocktree.hpp [hunk 2]
```diff
@@ -123,7 +123,7 @@ struct BlockTree : public BaseBlockTree<Block> {
                         pad);
   }
 
- protected:
+ public:
```
Comment (Warchant) [inline include/veriblock/blockchain/blocktree.hpp:126]: ?

File: include/veriblock/blockchain/base_block_tree.hpp (hunk 1 above)
Comment (blackyblack) [reply include/veriblock/blockchain/base_block_tree.hpp:196]: Temporary. Will revert soon.
